Cacher 2.0

rayzurbock opened the pull request:
- Simplifies code by removing repetitive echo (echo to screen + echo to file); replaces repetitive echo with tee as suggested by @hunty1 on PR 18 ( 2>&1 | tee -a "${tmplocation}"/AlertInfo.txt).  Previously, Cacher was 882 lines of code, this change brings it down to 744 and is easier to read and modify.
- Adds Uptime to emailed report.
- Adjusts Last Modified date from 05/16/2016 (inaccurate) to 9/23/2016.
- Adds a script version, randomly started at 2.0. I assume that's a good point to start incrementing the script version for all future PR's and updates.
 - Adds script version to the bottom of the emailed alert for reference when comparing reports.

erikng commented:
Would you be willing to backadd all the other iOS builds, OS X builds, sierra, Apple configuration and new iPad/iPhone models?

It's a decent amount of work but makes sense if we will officially call this 2.0

rayzurbock commented:
I thought about adding newer OS X versions including Sierra but I couldn't find a reliable source/database of identifiers (Darwin versions, etc).

I'll gladly add more if I can find a good resource or database to cross reference device to identifier.

---

Brian S. Lowrance

On Sat, Sep 24, 2016 at 12:48 AM -0500, "Erik Gomez" <notifications@github.com<mailto:notifications@github.com>> wrote:

Would you be willing to backadd all the other iOS builds, OS X builds, sierra, Apple configuration and new iPad/iPhone models?

It's a decent amount of work but makes sense if we will officially call this 2.0

## 

You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHubhttps://github.com/erikng/Cacher/pull/21#issuecomment-249347213, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AIBAqYunL6FyBQSRnomdW0ILY94szYVpks5qtLmugaJpZM4KFMJe.

rayzurbock commented:
Updated my 2.0 branch with the following:
- Updated 2.0 (Never merged; update 1 to 2.0)
- Additional code/comment cleanup
- Added code for iOS 10.0.2 detection
- Added code for iPhone 7, iPhone 7 Plus detection
- Added code for OS X detection from 10.5 to 10.9.5, 10.11.6 and 10.12.0. 
  - Apple OS X running 10.8.2 and above automatically use a nearby caching server if available; Uncomment previous versions if you want them in your reports (10.5 - 10.8.1)

erikng commented:
I'm confused on the older OS support. Can you explain this?

erikng commented:
Also looks like you have some "?" For the iPhone 7 models. Like what I'm seeing though.

rayzurbock commented:
You asked for older OS X versions, so I added them, but then afterwards I looked up info on the Apple Caching Server and it's supported clients to find that OS X clients didn't automatically start using a Caching server until 10.8.2, so to prevent giving a bunch of useless data that may likely be all counts of 0, I went back and commented out the versions earlier than 10.8.2 that I had added.  I left them in the code for users to uncomment if they wish because I didn't see anything that said that they were not supported, just that they weren't automatically found/used by the client OS.  If you want them, I'll remove the comments.

Regarding the iOS 7 question marks.  I'll put whatever is needed there, but I don't really know what those detections signify.  I went off of your earlier examples for those. Some (iPhone x,2 and x,4) were [GSM], [CDMA], and/or [China Model].  I couldn't find a good reference to tell me that iPhone9,1 = US iPhone7, iPhone9,2 = China,GSM,CDMA iPhone7, iPhone9,3= US iPhone7 Plus, iPhone9,4 = China,GSM,CDMA iPhone7 Plus.  Honestly, I guessed that iPhone9,1 and iPhone9,3 were the US versions of iPhone7 and iPhone7 Plus based on your previous examples and didn't really know if iPhone9,2 and iPhone9,4 = China, GSM, CDMA or other iPhone7 versions so I put a ?.  If anyone can confirm, I'll gladly replace the ?.

erikng commented:
Sorry for it being clear Brian. By older OS's I only meant the sub versions of OS X that hadn't been done and the iOS ones (9.3.5 and lower). We don't need anything prior to versions I supported. It will just result in a ton of 0's.

If you could, just make sure all of iOS 9/10 is handled, remove those Older OS X calls and I'll merge it. I can fix the ? As well.

Sent from my iPhone

On Sep 28, 2016, at 8:41 PM, Brian Lowrance <notifications@github.com<mailto:notifications@github.com>> wrote:

You asked for older OS X versions, so I added them, but then afterwards I looked up info on the Apple Caching Server and it's supported clients to find that OS X clients didn't automatically start using a Caching server until 10.8.2, so to prevent giving a bunch of useless data that may likely be all counts of 0, I went back and commented out the previous versions I had added. I left them in there because I didn't see anything that said that they were not supported, just that they weren't automatically found/used. If you want them, I'll remove the comments.

Regarding the iOS 7 question marks. I'll put whatever is needed there, but I don't really know what those detections signify. I went off of your earlier examples for those. Some (iPhone x,2 and x,4) were [GSM], [CDMA], and/or [China Model]. I couldn't find a good reference to tell me that iPhone9,1 = US iPhone7, iPhone9,2 = China,GSM,CDMA iPhone7, iPhone9,3= US iPhone7 Plus, iPhone9,4 = China,GSM,CDMA iPhone7 Plus. Honestly, I guessed that iPhone9,1 and iPhone9,3 were the US versions of iPhone7 and iPhone7 Plus based on your previous examples and didn't really know if iPhone9,2 and iPhone9,4 = China, GSM, CDMA or other iPhone7 versions so I put a ?. If anyone can confirm, I'll gladly replace the ?.

## 

You are receiving this because you commented.
Reply to this email directly, view it on GitHubhttps://github.com/erikng/Cacher/pull/21#issuecomment-250363031, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFa_GOwG2tWrlzTRbBg9MaDJQz7-zJHtks5quzNfgaJpZM4KFMJe.

rayzurbock commented:
So just to confirm:
- The only OSX versions that you wanted added to your version were 10.11.6 (El Capitan) and 10.12.0 (Sierra) right?
  - You want 10.8.2 - 10.10.5 removed from the code?  (Note: I have 23 clients reported on 10.10.5, but 0's for all previous).
- You want 9.3.3 - 9.3.5 and 10.0.2 added for iOS detection/reporting.

erikng commented:
If you have clients reporting go ahead and keep it.

And yes that all sounds great. :)

Sent from my iPhone

On Sep 28, 2016, at 8:56 PM, Brian Lowrance <notifications@github.com<mailto:notifications@github.com>> wrote:

So just to confirm:
-   The only OSX versions that you wanted added to your version were 10.11.6 (El Capitan) and 10.12.0 (Sierra) right?
  -   You want 10.8.2 - 10.10.5 removed from the code? (Note: I have 23 clients reported on 10.10.5, but 0's for all previous).
-   You want 9.3.3 - 9.3.5 and 10.0.2 added for iOS detection/reporting.

## 

You are receiving this because you commented.
Reply to this email directly, view it on GitHubhttps://github.com/erikng/Cacher/pull/21#issuecomment-250364522, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFa_GHMp9qHFyF-L16lCXFx0AecsEL7oks5quzbegaJpZM4KFMJe.

rayzurbock commented:
Updated PR.
- Removed OS X detection for 10.5.0 - 10.10.4 that was added in 2.0 PR Update 1 (Desired scope misunderstood)
- iOS detection for 9.3.3, 9.3.4, 9.3.5 added

erikng commented:
Sorry I haven't merged this yet. I'll try to do it tomorrow.

erikng commented:
It's merged. I also fixed the iPhone labels, added iOS 10.1, Sierra 10.12.1 and a note about AC 2.3. 

Thanks again for the PR. Sorry it took so long.

rayzurbock commented:
Awesome. Thanks.

rayzurbock commented:
I hope that the simplified codebase encourages and helps with future development,

erikng commented:
Soon I will have caching servers globally so I will try and spend development time on this. I've been extremely busy writing chef code as of late.

On Oct 13, 2016, at 6:57 PM, Brian Lowrance <notifications@github.com<mailto:notifications@github.com>> wrote:

I hope that the simplified codebase encourages and helps with future development,

## 

You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHubhttps://github.com/erikng/Cacher/pull/21#issuecomment-253690291, or mute the threadhttps://github.com/notifications/unsubscribe-auth/AFa_GK0dx9nX1MQplyGewGQUyOqw8lVqks5qzuF_gaJpZM4KFMJe.
